docs/dev/pass_infra.rst

```diff
@@ -389,6 +396,103 @@ To allow other C++ modules to apply this pass, we declare a free function in
 
     TVM_DLL Pass FoldConstant();
 
+.. _pass_instrument_section_tag:
+
+Pass Instrument
```

Comment:
it seems like PassContext may be introduced after this section, no? Perhaps it makes sense to move it downward.

Reply:
I fixed wordings.
As for the order of introduction, It's a little embarrassing. I would like to put all Pass Instrument descriptions together and after `PassContext` introduction.
But the `pass_infra.txt` is divided to C++ backend and Python Frontend parts.
I tried to follow that. So the order of introduction becomes:

1. `PassContext` C++ backend 
2. `PassInstrument` C++ backend
3. `PassContext` Python frontend
4. `PassInstrument` Python frontend

The `PassContext` Python frontend is after `PassInstrument` C++ backend.

Reply:
ah i see. what do you think about documenting the PassInstrument first, since that's the developer-facing thing? TVM calls `InstrumentEnterPassContext`.

i think explaining the C++ thing first and then the Python thing is workable. i agree it's a bit tricky to work around the split. I don't think the Python docs need to be as extensive since they mostly wrap the C++ stuff

Reply:
It would be nice! But after trying and reading around, I find PassInstrument might require understanding PassContext first to know about what each instrument points means.
Perhaps we can split PassInstrument doc out once it becomes bigger (such as more and more built-in instruments are added, or new functionality added to provide more instrumentation possibility?)

I also tried to shrink Python docs a little. Thanks a lot for suggestions :)